pkg/module/filter.go

```diff
@@ -29,16 +29,24 @@ type Filter struct {
 //   -
 //   + github.com/a
 // will exclude all items from communication except github.com/a
-func NewFilter(filterFilePath string) *Filter {
+func NewFilter(filterFilePath string) (*Filter, error) {
+	// Do not return an error if the file path is empty
+	// Do not attempt to parse it as well.
+	if filterFilePath == "" {
+		return nil, nil
+	}
+
 	rn := newRule(Default)
 	modFilter := Filter{
 		filePath: filterFilePath,
 	}
 	modFilter.root = rn
 
-	modFilter.initFromConfig()
-
-	return &modFilter
+	err := modFilter.initFromConfig()
+	if err != nil {
+		return nil, err
+	}
+	return &modFilter, nil
```

Comment:
`return `&modfilter,err` should be ok, in case it's nil it's nil

Reply:
Hmm.. In this case it is not nil since we declare modFilter above.

Reply:
first value should not matter if you have an error set. 
usually you do: if err { log/panic/return err }